packages/styles/scss/components/dialog/_dialog.scss

```diff
@@ -13,130 +13,171 @@
 @use '../../spacing' as *;
 @use '../../theme' as *;
 @use '../../type' as *;
-@use '../../utilities/ai-gradient' as *;
 @use '../../utilities/convert';
 @use '../../utilities/component-reset';
 @use '../../utilities/focus-outline' as *;
 @use '../../utilities/high-contrast-mode' as *;
+@use '../../utilities/update_fields_on_layer' as *;
+@use '../../utilities/z-index' as *;
 
 /// Dialog styles
 /// @access public
 /// @group dialog
 @mixin dialog {
   .#{$prefix}--dialog {
+    position: fixed;
+    z-index: z('modal');
+    display: flex;
+    align-items: center;
+    justify-content: center;
     padding: 0;
     border: none;
-    background-color: $layer;
-    block-size: fit-content;
+    background-color: transparent;
+    block-size: 100vh;
     color: $text-primary;
-    inline-size: 48rem;
-    max-block-size: 50%;
-    max-inline-size: 100%;
-    opacity: 0;
-    transform: translateY(calc(-1 * #{$spacing-06}));
-
-    /** opening and closing is used in as allow-discrete is not currently supported wide enough
-     * https://caniuse.com/mdn-css_properties_display_is_transitionable
-     */
-    transition:
-      opacity $duration-moderate-02 motion(exit, expressive),
-      transform $duration-moderate-02 motion(exit, expressive),
-      overlay $duration-moderate-02 motion(exit, expressive) allow-discrete,
-      display $duration-moderate-02 motion(exit, expressive) allow-discrete;
```

Comment:
Usage of `allow-discrete` was specifically included in the original implementation, we should keep this approach. Did you run into an issue with it?